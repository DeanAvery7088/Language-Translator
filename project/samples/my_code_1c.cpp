#include <iostream>
#include "Matrix.h"
#include <math.h>
using namespace std;
int main ( ) {
matrix m (2,2);
for (int i = 0; i < 2; i++) {
for (int j = 0; j < 2; j++) {
*(m.access(i, j)) = (i * j); 
} }

int i;
int j;
for (i = 0; i <= 1; i++) { 
for (j = 0; j <= 1; j++) { 
*(m.access(i, j)) = 2 ;
}

}

matrix n (2,2);
for (int h = 0; h < 2; h++) {
for (int k = 0; k < 2; k++) {
*(n.access(h, k)) = (h * k); 
} }

int h;
int k;
for (h = 0; h <= 1; h++) { 
for (k = 0; k <= 1; k++) { 
*(n.access(h, k)) = 2 ;
}

}

*(n.access(0, 1)) = 3 ;
*(n.access(1, 1)) = 11 ;
*(m.access(0, 0)) = 22 ;
*(m.access(1, 1)) = 13 ;
matrix L (2,2);
for (int a = 0; a < 2; a++) {
for (int b = 0; b < 2; b++) {
*(L.access(a, b)) = (a * b); 
} }

int a;
int b;
int c;
for (a = 0; a <= 1; a++) { 
for (b = 0; b <= 1; b++) { 
*(L.access(a, b)) = 0 ;
}

}

L = (m * n);

cout << m;
cout << "\n";
cout << n;
cout << "\n";
cout << L;
}
