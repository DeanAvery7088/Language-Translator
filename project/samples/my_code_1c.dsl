/* Test program that multiplies two 2x2 matrices and prints the result */

main () { 
  matrix m [ 2 : 2 ] i : j = i * j ;
  int i ;
  int j ;
  repeat ( i = 0 to 1 ) {
    repeat ( j = 0 to 1) {
      m[i:j] = 2 ;
    }
  }

  matrix n [ 2 : 2 ] h : k = h * k ;
  int h ;
  int k ;
  repeat ( h = 0 to 1 ) {
    repeat ( k = 0 to 1) {
      n[h:k] = 2 ;
    }
  }
  n[0:1] = 3;
  n[1:1] = 11;
  m[0:0] = 22;
  m[1:1] = 13;
  matrix L [ 2 : 2 ] a : b = a * b ;
  int a ; 
  int b ;
  int c ;
  repeat ( a = 0 to 1 ) {
    repeat ( b = 0 to 1 ) {
	L[a:b] = 0 ; 		   
    }
  }
  L = m * n;

  //repeat (a = 0 to 1) {
  //  repeat (b = 0 to 1) {
  //    print (L[a:b]) ;
  //    print ( " ") ; 
  //  }
  //  print ( "\n" ) ; 
  //}

  print(m);
  print("\n");
  print(n);
  print("\n");
  print(L);

}



