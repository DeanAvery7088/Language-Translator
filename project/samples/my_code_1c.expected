2 2
22  2  
2  13  

2 2
2  3  
2  11  

2 2
48  88  
30  149  
